server.js

```diff
@@ -4040,6 +4040,81 @@ cache(function(data, match, sendBadge, request) {
       badgeData.text[1] = 'invalid';
       sendBadge(format, badgeData);
     }
+   });
+}));
+
+function nexusRequest(data, match, sendBadge, request, urlCallback, dataCallback) {
+  var scheme = match[1]; // http(s)
+  var host = match[3]; // nexus.example.com
+  var groupId = match[4]; // eg, `com.google.inject`
+  var artifactId = match[5]; // eg, `guice`
+  var format = match[6] || "gif";
+  var apiUrl = scheme + '://' + host + urlCallback(encodeURIComponent(groupId), encodeURIComponent(artifactId));
+  var badgeData = getBadgeData('nexus', data);
+  request(apiUrl, { headers: { 'Accept': 'application/json' } }, function(err, res, buffer) {
+	 if (err != null) {
+	   badgeData.text[1] = 'inaccessible';
+	   sendBadge(format, badgeData);
+	   return;
+	 }
+	 try {
+	   var parsed = JSON.parse(buffer);
+	   var version = dataCallback(parsed);
```

Comment:
Parameterizing this logic with a function makes this difficult to reason about. Could you pass a string parameter instead, and use an `if` statement?